projects/ngx-quill/src/lib/quill.service.ts

```diff
@@ -74,4 +66,34 @@ export class QuillService {
   getQuill() {
     return this.quill$
   }
+
+  /**
+   * Marked as internal so it won't be available for `ngx-quill` consumers, this is only
+   * internal method to be used within the library.
+   *
+   * @internal
+   */
+   async registerCustomModules(
+    Quill: any,
+    customModules: CustomModule[] | undefined,
+    suppressGlobalRegisterWarning?: boolean
+  ): Promise<any> {
+    if (Array.isArray(customModules)) {
+      // eslint-disable-next-line prefer-const
+      for (let { implementation, path } of customModules) {
+        // The `implementation` might be an observable that resolves the actual implementation,
+        // e.g. if it should be lazy loaded.
+        if (isObservable(implementation)) {
+          // Note: switch to `firstValueFrom` once `toPromise()` is removed and we drop RxJS@6 support.
+          // The `firstValueFrom` is available only in RxJS@7.
+          implementation = await implementation.toPromise()
```

Comment:
since i will release this only for angular 14 with rxjs ^7, you can use firstValueFrom

Reply:
feel free to update it in your pr :)

Reply:
i just checked the dev deps and not the peer deps. since i planned not to support angular 13 with ngx-version >= 17 it is okay to bump the peer deps